### PR TITLE
Adds support for non-1:1 screen ratios (Doesn't add widescreen)

### PR DESCRIPTION
**Pull request by deathride58**
In layman's terms, it's a step towards paving the way for this:
image
# disclaimer: this image is an example, this PR doesn't add widescreen, but it does make adding it easier

To test/experiment with it on a live server, mass varedit client.view to "21x15", or any other non-1:1 viewscale.

For other devs: This PR introduces the getviewsize proc. This proc will attempt to return a consistent view size format from whatever's passed to it. This paves the way for non-1:1 aspect ratio support without affecting the existing 1:1 screen scale.

Goonstation has already adopted widescreen for living players, and it's a pretty huge quality of life improvement for players and devs alike. Devs get more room on the screen to place HUD objects without things looking as cluttered, players don't have a quarter of their screen unused in the status area, and the entire screen in general feels a lot less cluttered with unused/useless space.

### CODE REVIEW

**Comment by praisenarsie** [comment]
hoooooooooooooooooooo

**Comment by DoctoreMad** [comment]
This is a pretty huge disservice to people with small monitors. Also, it does a change to fighting because unrobust players are going to feel like west/east fighting is more "convenient" than north/south, something which might be palpable if two forces with ranged attacks are fighting.

This isn't just a "cover more of your screen change". Tread carefully.

**Comment by DoctoreMad** [comment]
Dev side it doesn't matter though, so I guess it can be added as debug stuff?

**Comment by ShizCalev** [comment]
21x9 master race

**Comment by Qbopper** [comment]
is there a significant number of people who still only have a 4:3 display, though?

would be nice if there were stats for that sort of thing

**Comment by deathride58** [comment]
This PR doesn't actually make the game widescreen, it just helps pave the way to allow a switch to widescreen in the future, should other contributors or downstreams choose to do so. 

As for the UI, there's currently not much room to place extra buttons without adding extra clutter, see Bay's aim toggle button for example, which makes it a little hard to see things in the bottom right corner of the screen. If Bay had widescreen, the aim toggle button could be moved to the bottom of the screen, reducing the amount of clutter in the bottom right corner.

For folks still on 4:3 monitors, it would be pretty easy to do what Goon did, and add an extra UI skin to switch between the chat/stat panel being on the right side of the screen or the bottom of the screen.

**Comment by Qbopper** [comment]
yeah I was just curious, I know what the PR does, I was talking if there's a hypothetical push towards widescreen support it would be nice to have stats on monitor sizes

your image is likely going to mislead people who don't read though, not much we can do about that

**Comment by deathride58** [comment]
>"Your image is likely going to mislead people who don't read"

fix'd (hopefully)

**Comment by DoctoreMad** [comment]
Well then.

**Comment by psykzz** [comment]
widescreen when

**Comment by deathride58** [comment]
Anyone with +varedit permissions can enable widescreen for the entire server or for specific players, regardless if this PR gets merged or not. Proper widescreen support will require fullscreen sprites updated for the 7:5 aspect ratio. I've yet to hunt down every single proc that relies upon the view var, but until they're updated to take into account the possibility of view being a string instead of a num, they'll probably cause runtimes, which could potentially break things.

**Comment by Cyberboss** [comment]
Travis

**Comment by deathride58** [comment]
image

image

This is gonna take a while to get right

**Comment by deathride58** [comment]
Doing some tests, it appears the only issues remaining are silicons causing runtimes when they're set to a non-1:1 view, guns resetting the client viewrange when dropped (out of the PR's scope), non-1:1 ghosts causing runtimes when trying to view tgui objects, and the clickcatcher not updating when view is changed.

Everything else seems to work perfectly, with 1:1 views and 7:5 views both functioning for humans without any noticeable issues. After I resolve the rest of the existing issues, adding widescreen would be as simple as finding a way to make the view variable persist between reconnects and other things that reset view, but that's outside of the scope for this PR.

**Comment by Cyberboss** [comment]
Remove the [WIP] when you think its ready

**Comment by optimumtact** [comment]
I enjoy this pr because it will make a subclass of poor people who will get robusted

support :+1:

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -651,6 +651,9 @@ GLOBAL_LIST(external_rsc_urls)
 			return FALSE
 		if ("key")
 			return FALSE
+		if("view")
+			change_view(var_value)
+			return FALSE
```
Comment (Cyberboss) [inline code/modules/client/client_procs.dm]: This needs to return TRUE, otherwise it says "Your edit was rejected by the object!"

**Review by Cyberboss — DISMISSED** [review]

**Comment by Cyberboss** [comment]
```
proc name: update o (/obj/screen/parallax_layer/proc/update_o)
usr: Cassian Rifler (evasivepotato) (/mob/dead/observer)
usr.loc: The floor (82,122,2) (/turf/open/floor/wood)
src: (/obj/screen/parallax_layer/layer_1)
src.loc: null
call stack:
(/obj/screen/parallax_layer/layer_1): update o("21x15")
/datum/hud/ghost (/datum/hud/ghost): update parallax()
Cassian Rifler (/mob/dead/observer): update parallax contents()
Cassian Rifler (/mob/dead/observer): Move(the floor (82,122,2) (/turf/open/floor/wood), 2)
Evasive Potato (/client): Move(the floor (82,122,2) (/turf/open/floor/wood), 2)
Evasive Potato (/client): .movedown()
```

**Comment by Cyberboss** [comment]
This seems fine, remove the debug code and I'll merge it
